Extractor/extractor.cpp

```diff
@@ -22,17 +22,57 @@ int main(int argc, char *argv[])
       .help("Output folder where all files will be extracted")
       .default_value(std::filesystem::current_path().string());
 
+  program.add_description(
+      "For more information please visit "
+      "https://github.com/wagrenier/Mikompilation");
+
   try
   {
     program.parse_args(argc, argv);
-    Extractor(program.get("iso").c_str(), program.get("output").c_str());
+
+    isoFile = program.get("iso");
+    outputDirectory = program.get("output");
   }
   catch (const std::runtime_error &err)
   {
-    std::cerr << err.what() << std::endl;
     std::cerr << program;
     std::exit(1);
   }
 
+  IsoReader isoReader(isoFile.string());
+  ZeroReader *zeroReader = nullptr;
+
+  if (!isoReader.ValidGameRegion())
+  {
+    std::cerr << program;
+    return 0;
+  }
+
+  switch (isoReader.GetGameType())
```

Comment:
this switch case should not be in the main function, it should be contained in a seperate file, hence why initially I had the main file and the extractor file. Please but it back as it was :)